src/config_brain_memory.js

```diff
@@ -188,7 +188,7 @@ Upgrade less: ${strings.upgradeLess}
 };
 
 brain.prepareMemory = function() {
-  Memory.username = Memory.username || _.find(Game.spawns, 'owner').owner.username;
+  Memory.username = Memory.username || _.chain(Game.rooms).map('controller').flatten().filter('my').map('owner.username').first().value();
```

Comment:
SYSTEM_USERNAME is "Screeps"

Regarding performance panic over using chain, search for "lodash shortcut fusion" for example: http://gajus.com/blog/4/harder-better-faster-stronger-lo-dash-v3#shortcut-fusion